packages/headless/src/features/question-answering/question-answering-slice.ts

```diff
@@ -25,4 +42,33 @@ export const questionAnsweringReducer = createReducer(
         state.liked = false;
         state.disliked = true;
       })
+      .addCase(executeSearch.fulfilled, (state, action) => {
+        state.relatedQuestions = action.payload.response.questionAnswer.relatedQuestions.map(
+          (relatedQuestion) => ({
+            contentIdKey: relatedQuestion.documentId.contentIdKey,
+            contentIdValue: relatedQuestion.documentId.contentIdValue,
+            expanded: false,
+          })
+        );
+      })
+      .addCase(expandSmartSnippetRelatedQuestion, (state, action) => {
+        const idx = findRelatedQuestionIdx(
+          state.relatedQuestions,
+          action.payload as QuestionAnswerDocumentIdentifier
```

Comment:
Why is the cast needed? Same for line 67.

Reply:
It's due to the type definition coming in from rtk.